### DISCUSSION TITLE
Components should suck less

### DISCUSSION THREAD

**Discussion by mygamingaccount** [FakeFishGames/Barotrauma#10107] [Ideas and suggestions]
The ![image](https://user-images.githubusercontent.com/68995233/196350907-e329b195-0e13-4f24-8f4a-9d568348d2c2.png), ![image](https://user-images.githubusercontent.com/68995233/196350964-60973209-1a42-4fbe-9479-072eb04ca178.png), ![image](https://user-images.githubusercontent.com/68995233/196351018-5c3bdfe6-8ca7-4c7e-b83a-2374d8170311.png), ![image](https://user-images.githubusercontent.com/68995233/196351079-ffa2cd4d-cacf-4de8-a49a-867789e2c4c8.png) arithmetic components should be able to store a constant value, the value on the right side of the operator.

![image](https://user-images.githubusercontent.com/68995233/196351359-768656c7-1306-4a87-a520-b2a5d06703c7.png) and the Equals components should store a value that the incoming signal is compared against.

Rounding should also have a parameter of the specific number of decimals points to be rounded to

This would cut down on the use of memory components, and improve performance by not having to convert between strings and floats for every other input.

I have no doubt that these changes are necessary. For example, people have been using RegEx components instead of GreaterThan, such is the hatred against having to use Memory components

**Comment by gerakis** [comment]
I would like it if they also allowed you to store a constant value for either of the left inputs also. Sort of like how output works on most components. You can set it manually, but also override it with a pin. This would eliminate the mem components in most forms of static math, like some input ![196351018-5c3bdfe6-8ca7-4c7e-b83a-2374d8170311](https://user-images.githubusercontent.com/16128947/196748087-66f7205f-13af-4fd2-9b18-0c1e1a7cd523.png) 0.1 for instance.

**Reply by mygamingaccount** [reply]
It would also let you have a ![image](https://user-images.githubusercontent.com/68995233/196351018-5c3bdfe6-8ca7-4c7e-b83a-2374d8170311.png) output a constant value, the product of the numbers you've preset it with. I'm not sure that's the right direction.

Timeframe further complicates this as that would either have to be disabled when a constant value is loaded, or the the implementation has to be even more convoluted than it already is: it has to wait for either inputs to receive a signal. Maybe its the best to just remove the timeframe parameter, since it would be technically infinite, with the incoming values being stored.

**Reply by mygamingaccount** [reply]
Timeframe in boolean logic components has shown to be useful in many cases, unlike the arithmetic components.
